File: lib/models/subject.dart
import 'package:flutter/material.dart';

class Subject {
  final String id;
  final String name;
  final String englishName;
  final IconData icon;
  final Color color;
  final String description;

  const Subject({
    required this.id,
    required this.name,
    required this.englishName,
    required this.icon,
    required this.color,
    required this.description,
  });
}

File: backend/app/Models/Subject.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Subject extends Model
{
    use HasFactory;

    protected $fillable = [
        'code',
        'name',
        'english_name',
        'icon',
        'color',
        'description',
        'sort',
        'is_active',
    ];

    protected $casts = [
        'is_active' => 'boolean',
        'sort' => 'integer',
    ];

    public function lessons(): HasMany
    {
        return $this->hasMany(Lesson::class);
    }

    public function questions(): HasMany
    {
        return $this->hasMany(Question::class);
    }
}
